### PR TITLE
CA-257: Handle case of access policies on managed groups when publishing sync messages

### PR DESCRIPTION
**Pull request by ansingh7115**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

**Comment by ansingh7115** [comment]
jenkins retest

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 1]
```diff
@@ -190,13 +193,15 @@ class GoogleExtensions(
...
-          case FullyQualifiedPolicyId(FullyQualifiedResourceId(ManagedGroupService.managedGroupTypeName, id: ResourceId),
-                                                 ManagedGroupService.adminPolicyName | ManagedGroupService.memberPolicyName) =>
-            makeGroupPublishMessages(WorkbenchGroupName(id.value))
-          // if it's an access policy on a resource that's not a managed group
+          // it is a group that isn't an access policy, could be a managed group
+          case groupName: WorkbenchGroupName =>
+            makeConstrainedResourceAccessPolicyMessages(groupName).map(_  :+ groupName.toJson.compactPrint)
+
+          // it is the admin or member access policy of a managed group
+          case accessPolicyId@FullyQualifiedPolicyId(FullyQualifiedResourceId(ManagedGroupService.managedGroupTypeName, id: ResourceId), ManagedGroupService.adminPolicyName | ManagedGroupService.memberPolicyName) =>
```
Comment (gpolumbo-broad) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: Is there any way we can simplify this line or break it up or something?

**Review by gpolumbo-broad — APPROVED** [review]

**Review by marctalbott — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala (hunk 1 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala]: I made it  a little shorter but it is a pattern match so I don't know what else to do

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest please

**Comment by dvoet** [comment]
jenkins retest

**Pull request merged**